sklearn/utils/validation.py

```diff
@@ -1259,3 +1271,34 @@ def inner_f(*args, **kwargs):
         kwargs.update({k: arg for k, arg in zip(all_args, args)})
         return f(**kwargs)
     return inner_f
+
+
+def _check_fit_params(fit_params):
+    """Check and validate the parameters passed during `fit`.
+
+    Parameters
+    ----------
+    fit_params : dict
+        Dictionary containing the parameters passed at fit.
+
+    Returns
+    -------
+    fit_params_validated : dict
+        Validated parameters. We ensure that the values support indexing.
+    """
+    fit_params_validated = {}
+    for param_key, param_value in fit_params.items():
+        is_scalar = [
+            check(param_value)
+            for check in [np.isscalar, ismodule, isclass, ismethod, isfunction]
+        ]
+        if any(is_scalar):
+            # keep scalar as is for backward-compatibility
+            # https://github.com/scikit-learn/scikit-learn/issues/15805
+            fit_params_validated[param_key] = param_value
```

Comment:
Instead of checking for a list of authorized type families (`[np.isscalar, ismodule, isclass, ismethod, isfunction]`), we could just rely on `_is_arraylike` as is also done in the `_index_param_value` function used in `_fit_and_score`:

```python
if not _is_arraylike(param_value):
    # Non-indexable pass-through (for now).
    fit_params_validated[param_key] = param_value
else:
    fit_params_validated[param_key] = _make_indexable(param_value)
```

Reply:
Thanks I did not know about this helper.